### ISSUE TITLE
Assigning static IP to windows VM residing on ESX server

### ISSUE THREAD

**Issue by dipaligulhane** [vmware/govmomi#1587]
Hi,
Not sure if this is correct way to raise queries here, if there is other way of doing so please let me know.

I need to set static IP to newly deployed windows VM. Currently using this library to get all VM properties, however not sure how to set IP to VM on esx host. 
Got this link https://github.com/vmware/govmomi/tree/928976b41049dede3336b43e82e50e0351fdd7f0/toolbox however supported for Linux VM as per documentation.
Can someone help me with pointers to example of adding static IP to windows VM?

**Comment by dougm** [comment]
@gressd may have some pointers here.

**Comment by gressd** [comment]
Hi @dipaligulhane

Coming soon if tests and code review passes !!
      https://github.com/vmware/govmomi/pull/1576

**Comment by dipaligulhane** [comment]
@gressd went through your changes and have few queries. Code looks like to use govc command line options to set IP. 
I am using this library in Go code for automation. I am not using govc command line utility.
and I am not getting how can I use these customization options while using library. Would appreciate your inputs on the same.

**Comment by AndrewDi** [comment]
@dougm it seems like I can use `govc vm.customize -vm VM -gateway GATEWAY -netmask NETMASK -ip NEWIP -dns-server DNS1 -dns-server DNS2 NAME` to assign new ip to `win/linux` vm, but our vm has multi network adapter, is there any way to update specific network adapter?

**Comment by dougm** [comment]
@AndrewDi we could support multiple `-ip` args, will look at that before the 0.22 release.

**Comment by dougm** [comment]
This is a little tricky with the way Go's flags work, esp. since `-gateway` and `-dns-server` are already multiple arg, per-adapter.  We could add an optional index within the flag values, to indicate which adapter they apply to.  For example:

```console
% govc vm.customize -vm VM -ip 10.0.0.20@1 -gateway 10.0.0.1@1 -ip 10.20.0.10@2 -gateway 10.20.0.1@2
```

Where `@N` is the adapter index.  Thoughts or other suggestions?

**Comment by AndrewDi** [comment]
@dougm Very well, but how can I specific the network mapping?

**Comment by dougm** [comment]
Can you expand on what you mean by network mapping?

**Comment by AndrewDi** [comment]
My misunderstanding, `@1` is the first network adapter,`@2` is the second network adapter...?

**Comment by dougm** [comment]
@AndrewDi see #1771 , added support for multiple `-ip` flags. Rather than the @index idea, added optional `-mac`. Can you review?

**Comment by dougm** [comment]
@AndrewDi merged #1771 , feedback still welcome

@dipaligulhane govmomi example added in #1731

**Comment by jremerich** [comment]
I'm sorry to reopen this issue, but I'm having an error on assing a static IP to a VM.

Here's my command line (I'm using in docker):
`docker run -e GOVC_URL=https://user:pass@esxi_server -e GOVC_INSECURE=true --rm -it vmware/govc /govc vm.customize -vm vm-name -mac 00:0c:29:7b:01:d5 -ip 1.1.1.1 -netmask 255.255.255.0 -type=Linux`

Here some anothers shots:
- `govc vm.customize -vm vm-name -mac 00:0c:29:7b:01:d5 -ip 1.1.1.1 -netmask 255.255.255.0`
- `govc vm.customize -vm vm-name -mac 00:0c:29:7b:01:d5 -ip 1.1.1.1 -netmask 255.255.255.0 Linux`
- `govc vm.customize -vm vm-name -name new-name`

I can run anothers commands like `govc vm.ip -a -v4 vm-name` and more. But some commands returns the same message: `/govc: The operation is not supported on the object.`

My VM's OS are Centos 7 and Centos 8.

May you guys helps me?

Thanks!
